tests/sentry/api/endpoints/test_group_details.py

```diff
@@ -95,6 +98,74 @@ def test_no_releases(self):
         assert response.data["firstRelease"] is None
         assert response.data["lastRelease"] is None
 
+    def test_current_release(self):
+        clock = MockClock()
+
+        # Create several of everything, to exercise all filtering clauses.
+
+        def set_up_organization():
+            organization = self.create_organization()
+
+            team = self.create_team(organization=organization)
+            self.create_team_membership(team=team, user=self.user)
+
+            prod = self.create_organization_environment(organization, name="production")
+            dev = self.create_organization_environment(organization, name="development")
+
+            def set_up_project(environment):
+                project = self.create_project(organization=organization, teams=[team])
+                environment.add_project(project)
+
+                def set_up_group():
+                    group = self.create_group(project=project)
+
+                    def set_up_release():
+                        release = self.create_release(project=project)
+
+                        timestamp = clock()
+                        ReleaseEnvironment.get_or_create(project, release, environment, timestamp)
+                        group_release = GroupRelease.get_or_create(
+                            group, release, environment, timestamp
+                        )
+
+                        return group_release
+
+                    set_up_release()
+                    latest_gr = set_up_release()
+
+                    return group, latest_gr
+
+                set_up_group()
+                target_group, target_gr = set_up_group()
+
+                # TODO(ryanskonnord): We should be able to call set_up_group()
+                # once more here and still get the target GroupRelease, but it
+                # get None instead. This may be exposing an implementation bug?
```

Comment:
Any input on this?

My hunch is that the code under test is filtering by an ordered result set of `ReleaseEnvironment` rows, and then attempting to match a `GroupRelease` row by matching its `release_id` to that of the most recent `ReleaseEnvironment`. If there is a `ReleaseEnvironment` with a later `first_seen` belonging to a different group, but the same project/org/env, we erroneously filter down to that one row as "the latest", then filter down to zero because its group ID doesn't match.

But I'd like confirmation from someone more familiar with the intent of the code.

Reply:
I think @wedamija is familiar with this query as it has caused several problems in the past.

Reply:
@markstory I don't specifically remember anything about the intent of this query, although I've dealt with performance things around `ReleaseEnvironment` before. Usually on large orgs that have thousands of rows in the table.

@RyanSkonnord I find this comment pretty confusing
```
            # the current release is the 'latest seen' release within the
            # environment even if it hasnt affected this issue
```
It seems like it wants to find the latest seen release regardless of the group. But then I'm not sure why it filters by group id if that's the case. Where do we actually use this field in the UI? I assume it's something to do with filtering by environment?

Reply:
> It seems like it wants to find the latest seen release regardless of the group. But then I'm not sure why it filters by group id if that's the case.

Oh yeah, good point. The comment suggests that the part where it filters down to the latest `ReleaseEnvironment` row from another group is working as intended. But then, filtering down to group ID (to your point) causes it to return `None` rather than the later release. So that probably is a bug, though a different one than I first interpreted.

I think the comment justifies having two test cases -- one where the latest release is associated with the queried group ID and one where it isn't. Although that would seem to be a breaking test in the current state.

> Where do we actually use this field in the UI? I assume it's something to do with filtering by environment?

As a symptom of having my head buried in the back-end concerns for too long at a time, I don't know. High time to go find out.